src/hooks/useUserSettings.ts

```diff
@@ -0,0 +1,30 @@
+import { useAppDispatch, useAppState } from '../store';
+import { UserSettingsAction } from '../store/actions';
+import { UserSettings } from '../store/types';
+
+type AvailableSettings = keyof UserSettings;
+
+const settingsUpdateActions: Record<AvailableSettings, UserSettingsAction['type']> = {
+  preferredCurrencyCode: 'SET_PREFERRED_CURRENCY'
+};
+
+export const useUserSettings = (settingName?: AvailableSettings) => {
+  const { userSettings } = useAppState();
+  const dispatch = useAppDispatch();
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const updateSetting = (setting: AvailableSettings, value: any) => {
+    const type = settingsUpdateActions[setting];
+    const action = {
+      type,
+      payload: value
+    };
+
+    dispatch(action);
+  };
+
+  return {
+    value: settingName ? userSettings[settingName] : userSettings,
```

Comment:
can this be used now to get multiple settings out of this hook?

Reply:
i believe not right? If so, maybe an idea to spread the `userSettings` state here.

Reply:
you then also need to pass the `settingsName` to the hook.

Reply:
But I think I get your meaning. I wanted the return `value` contained in a single field, so as not to overcrowd the return object's fields

Reply:
Cool! Will do